### PR TITLE
Add network_response to BoxOAuthException

### PR DESCRIPTION
**Pull request by anthonywee**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @anthonywee has signed the CLA. Thanks for the pull request!

**Comment by anthonywee** [comment]
This is a PR to compare 2 options for allowing for retries of a failed downscope_token() request. The other PR is #247

File: test/unit/test_exception.py [hunk 1]
```diff
@@ -47,14 +47,17 @@ def test_box_oauth_exception():
...
         url=url,
         method=method,
+        network_response=network_response,
     )
     assert str(box_exception) == '''
 Message: {0}
 Status: {1}
 URL: {2}
 Method: {3}'''.format(message, status, url, method)
+    assert box_exception.network_response == network_response
```
Comment (jmoldow) [inline test/unit/test_exception.py]: ` is ` would be more appropriate than ` == ` here.

File: boxsdk/exception.py [hunk 2]
```diff
@@ -193,12 +193,28 @@ def __init__(self, status, message=None, url=None, method=None):
...
         self._url = url
         self._method = method
+        self._network_response = network_response
+
+    @property
+    def network_response(self):
+        """The response for the request.
+
+        :return network_response:
+            The network response for the request.
```
Comment (jmoldow) [inline boxsdk/exception.py]: This is exactly the same as what is said one line above. I don't think you need `:return:` for this.

**Comment by jmoldow** [comment]
Can you also add `headers`, and include `headers` in the string representation?

**Comment by jmoldow** [comment]
Please update the `HISTORY` file.

**Comment by anthonywee** [comment]
For `headers`, do you mean the response headers or the request headers? For the response headers, that should be accessible from the `network_response`. Would we expose that as a convenience?

**Comment by jmoldow** [comment]
I guess it doesn't need to be a separate param or a separate property. But can you put it into the message?

**Pull request merged**